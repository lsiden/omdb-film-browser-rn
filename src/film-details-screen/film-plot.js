import React from "react"
import PropTypes from "prop-types"
import { Text } from "react-native"
import { connect } from "react-redux"

import { detailExists } from "./detail-items"
import style from "./style"

export const filmPlot = ({ plot }) => {
  if (!detailExists(plot)) {
    return <Text style={style.plotUnavailable}>Plot unavailable</Text>
  }
  return <Text style={style.plot}>{plot}</Text>
}

filmPlot.propTypes = {
  plot: PropTypes.string
}

export default connect(state => ({
  plot: (state.filmDetails || {}).Plot
}))(filmPlot)
