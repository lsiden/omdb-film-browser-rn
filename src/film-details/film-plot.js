import React from "react"
import PropTypes from "prop-types"
import { Text } from "react-native"

import { detailExists } from "./detail-items"
import { detailStyles } from "styles"

const FilmPlot = ({ filmDetails }) => {
  const { plot } = filmDetails

  if (!detailExists(plot)) {
    return <Text style={detailStyles.plot}>Plot unavailable</Text>
  }
  return <Text>{filmDetails.Plot}</Text>
}

FilmPlot.propTypes = {
  filmDetails: PropTypes.object.isRequired,
}

export default FilmPlot
