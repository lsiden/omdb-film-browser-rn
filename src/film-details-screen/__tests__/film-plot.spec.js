import React from "react"
import renderer from "react-native-test-utils"

import { filmPlot as FilmPlot } from "../film-plot"

const filmDetails = require("__fixture__/film-details.json")

const defaultProps = () => {
  return {
    plot: filmDetails.Plot,
  }
}

const createWrapper = (props = {}) =>
  renderer(<FilmPlot {...{ ...defaultProps(), ...props }} />)

it("renders a title", () => {
  const wrapper = createWrapper()
  expect(wrapper.toJSON()).toMatchSnapshot()
})
