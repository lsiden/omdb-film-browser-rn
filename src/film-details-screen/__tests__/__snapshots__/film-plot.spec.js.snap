// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a title 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "color": "darkgray",
      "marginLeft": 20,
    }
  }
>
  In the 1950s, fear and violence escalate as the people of Algiers fight for independence from the French government.
</Text>
`;
